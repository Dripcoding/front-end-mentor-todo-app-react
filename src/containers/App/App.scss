@import "../../mixins";
@import "../../palette.scss";

$background-desktop-dark: url("../../assets/images/bg-desktop-dark.jpg");
$background-mobile-dark: url("../../assets/images/bg-mobile-dark.jpg");
$background-desktop-light: url("../../assets/images/bg-desktop-light.jpg");
$background-mobile-light: url("../../assets/images/bg-mobile-light.jpg");

.appContainer__mainContainer {
  @include flex(column, flex-start, center);
  width: 100%;
  height: 100vh;
  padding-top: 70px;
}

.background-desktop-dark {
  background-image: $background-desktop-dark;
  background-color: $very-dark-blue;
  background-repeat: no-repeat;
  background-size: 100% 300px;
}

.background-mobile-dark {
  background-image: $background-mobile-dark;
}

.background-desktop-light {
  background-image: $background-desktop-light;
}

.background-mobile-light {
  background-image: $background-mobile-light;
}
