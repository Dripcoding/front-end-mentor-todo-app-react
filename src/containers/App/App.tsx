import React from "react";
import Title from "../../components/Title/index";
import TodoInput from "../../components/TodoInput/index";
import TodoList from "../TodoList/index";

import "./App.scss";

const STYLE_BASE: string = "appContainer__";

const App = (): JSX.Element => {
  return (
    <main className={`${STYLE_BASE}mainContainer background-desktop-dark`}>
      <Title />
      <TodoInput />
      <TodoList />
    </main>
  );
};

export default App;
