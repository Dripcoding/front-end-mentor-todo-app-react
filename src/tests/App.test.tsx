import { render } from '@testing-library/react';
import App from '../containers/App/App';

describe("App", () => {
    it("matches snapshot", () => {
        const { container } = render(<App />);
        expect(container).toMatchSnapshot();
    })

    it("renders correctly", () => {
        const { getByRole, getByTestId } = render(<App />);

        const todoHeader = getByRole('heading');
        expect(todoHeader).toBeInTheDocument();
        expect(todoHeader).toBeVisible()
        
        const sunIcon = getByTestId('SUN-TOGGLE-ICON');
        expect(sunIcon).toBeInTheDocument();
        expect(sunIcon).toBeVisible();
    })
})
