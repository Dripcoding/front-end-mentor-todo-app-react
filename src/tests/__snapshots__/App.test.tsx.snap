// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div>
  <main
    class="appContainer__mainContainer background-desktop-dark"
  >
    <div
      class="title__container"
    >
      <h1>
        Todo
      </h1>
      <svg
        data-testid="SUN-TOGGLE-ICON"
      >
        icon-sun.svg
      </svg>
    </div>
    <div
      class="todoInput__container"
    >
      <div
        class="todoInput__input"
      >
        <div
          class="todoCheckBoxToggle__container"
          data-testid="TODO_CHECKBOX_TOGGLE"
        />
        <input
          class="todoInput__todoInput"
          placeholder="Create a new todo..."
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="todoList__container"
      data-testid="TODO_LIST_CONTAINER"
    >
      <div
        class="todoListItem__container"
        data-testid="TODO_LIST_ITEM"
      >
        <div
          class="todoCheckBoxToggle__container"
          data-testid="TODO_CHECKBOX_TOGGLE"
        />
        <input
          class="todoListItem__input"
          type="text"
          value=""
        />
      </div>
    </div>
  </main>
</div>
`;
